Inspect kwok tests

### What happened?

When running tests it seems some of them are likely to fail quite often:


 === RUN   TestKwokCluster/appsv1/deployment/pod_ready
    kwok_test.go:73: context deadline exceeded
--- FAIL: TestKwokCluster (120.02s)
    --- FAIL: TestKwokCluster/appsv1/deployment (120.02s)
        --- FAIL: TestKwokCluster/appsv1/deployment/deployment_creation (60.00s)
        --- FAIL: TestKwokCluster/appsv1/deployment/pod_ready (60.00s)
panic: interface conversion: interface {} is nil, not *v1.Deployment [recovered]
	panic: interface conversion: interface {} is nil, not *v1.Deployment
goroutine 45 [running]:
testing.tRunner.func1.2({0x20b1620, 0xc00020a000})
	/usr/local/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x650
panic({0x20b1620, 0xc00020a000})
	/usr/local/go/src/runtime/panic.go:890 +0x263
sigs.k8s.io/e2e-framework/examples/kwok/kwok_with_config.TestKwokCluster.func4({0x2582730, 0xc000586090}, 0xc00031c9c0, 0x2581b00?)
	/home/prow/go/src/sigs.k8s.io/e2e-framework/examples/kwok/kwok_with_config/kwok_test.go:78 +0x165
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).executeSteps(0xc00014f830, {0x2582730, 0xc000586090}, 0xc000586902?, {0xc000366220, 0x1, 0x2ce1120?})
	/home/prow/go/src/sigs.k8s.io/e2e-framework/pkg/env/env.go:428 +0x136
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).execFeature.func1(0xc00031c9c0)
	/home/prow/go/src/sigs.k8s.io/e2e-framework/pkg/env/env.go:486 +0x7ac
testing.tRunner(0xc00031c9c0, 0xc0002be1b0)
	/usr/local/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run === RUN   TestKwokCluster/appsv1/deployment/pod_ready
    kwok_test.go:73: context deadline exceeded
--- FAIL: TestKwokCluster (120.02s)
    --- FAIL: TestKwokCluster/appsv1/deployment (120.02s)
        --- FAIL: TestKwokCluster/appsv1/deployment/deployment_creation (60.00s)
        --- FAIL: TestKwokCluster/appsv1/deployment/pod_ready (60.00s)
panic: interface conversion: interface {} is nil, not *v1.Deployment [recovered]
	panic: interface conversion: interface {} is nil, not *v1.Deployment
goroutine 45 [running]:
testing.tRunner.func1.2({0x20b1620, 0xc00020a000})
	/usr/local/go/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x650
panic({0x20b1620, 0xc00020a000})
	/usr/local/go/src/runtime/panic.go:890 +0x263
sigs.k8s.io/e2e-framework/examples/kwok/kwok_with_config.TestKwokCluster.func4({0x2582730, 0xc000586090}, 0xc00031c9c0, 0x2581b00?)
	/home/prow/go/src/sigs.k8s.io/e2e-framework/examples/kwok/kwok_with_config/kwok_test.go:78 +0x165
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).executeSteps(0xc00014f830, {0x2582730, 0xc000586090}, 0xc000586902?, {0xc000366220, 0x1, 0x2ce1120?})
	/home/prow/go/src/sigs.k8s.io/e2e-framework/pkg/env/env.go:428 +0x136
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).execFeature.func1(0xc00031c9c0)
	/home/prow/go/src/sigs.k8s.io/e2e-framework/pkg/env/env.go:486 +0x7ac
testing.tRunner(0xc00031c9c0, 0xc0002be1b0)
	/usr/local/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run

### What did you expect to happen?

Tests execute properly everytime.

### How can we reproduce it (as minimally and precisely as possible)?

Run tests.

### Anything elese we need to know?

_No response_

### E2E Provider Used

kwok

### `e2e-framework` Version

latest

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

--- a/examples/kwok/kwok_with_config/kwok_test.go
+++ b/examples/kwok/kwok_with_config/kwok_test.go
@@ -68,7 +68,7 @@ func TestKwokCluster(t *testing.T) {
 			}
 			pod := pods.Items[0]
 			// wait for the pod to be ready
-			err = wait.For(conditions.New(cfg.Client().Resources()).PodConditionMatch(&pod, corev1.PodReady, corev1.ConditionTrue), wait.WithTimeout(time.Minute*1))
+			err = wait.For(conditions.New(cfg.Client().Resources()).PodConditionMatch(&pod, corev1.PodReady, corev1.ConditionTrue), wait.WithTimeout(time.Minute*3))
 			if err != nil {
 				t.Fatal(err)
 			}

